tests/Collector/SuperGlobalsCollectorTest.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tests\SensioLabs\Deptrac\Collector;
+
+use PHPUnit\Framework\TestCase;
+use SensioLabs\Deptrac\AstRunner\AstMap;
+use SensioLabs\Deptrac\Collector\DirectoryCollector;
+use SensioLabs\Deptrac\Collector\Registry;
+use SensioLabs\Deptrac\Collector\SuperGlobalsCollector;
+
+final class SuperGlobalsCollectorTest extends TestCase
+{
+    public function dataProviderSatisfy(): iterable
+    {
+        yield [['regex' => 'foo/layer1/.*'], __DIR__.'/Fixtures/SuperGlobals_contained.php', true];
+        yield [['regex' => 'foo/layer1/.*'], __DIR__.'/Fixtures/SuperGlobals_free.php', false];
```

Comment:
instead of all the mocking involved in the existing test-suite I wanted to bring this example up, with some test-fixture files.

IMO using concrete files would make the Test-suite easier to read, and new tests easier to write. also it has the benefit of actually testing what happens when the end user is using the tool instead of stubbed out mocks - which make the Tests hard to grasp.

see e.g. https://blog.frankdejonge.nl/testing-without-mocking-frameworks/